src/mpid/ch4/src/ch4_progress.c

```diff
@@ -14,6 +16,38 @@ static inline int progress_made(MPID_Progress_state * state)
     return (state->progress_count != state->progress_start);
 }
 
+static int get_progress_vci(MPID_Progress_state * state)
+{
+    int vci;
+
+    if (MPIDI_global.n_vcis == 1) {
+        /* single vci */
+        vci = 0;
+    } else if ((++state->poll_count) == MPIDI_CH4_GLOBAL_POLL_FREQ) {
```

Comment:
Sorry, I was reading the commits out of order. What I am referring to is the global progress case here. Why not make this the default case? This should be useful to evaluate when we actually have multiple VCIs enabled, if I understand correctly. The other cases are all optimizations.

Reply:
I see. I assume your motivation is to keep the ADI intact?

We still need the `MPID_Progress_state` to track `poll_count`. If we use global, it goes back to a global serialization. We could let go `poll_count` and poll every vci in every test, that not only means a global serialization, but also means we'll perform slower with multiple vci. For example, with 2 vci, 2 times slower, 4 vci 4 times slower, or more.

There isn't much point of getting the feature into a release that the feature will perform much worse even than global, right?

Reply:
Sigh. Perhaps I'm not making my point clear. I'm not rejecting the optimizations. I'm saying they cannot be evaluated in this state. It is not possible to test this code right now. I do not support hacking up the ADI just to merge this PR. It should be proposed along with the code that actually uses it.